share/steel/examples/pulse/bug-reports/Bug.Invariants.fst

```diff
@@ -0,0 +1,121 @@
+module Bug.Invariants
+open Pulse.Lib.Pervasives
+module U32 = FStar.UInt32
+
+```pulse
+atomic
+fn return_atomic
+      (x:ref U32.t)
+requires emp ** pts_to x 1ul
+returns n:U32.t
+ensures emp ** pts_to x 1ul
+{
+    read_atomic x;
+}
+```
+
+[@@expect_failure] //returns in Unobservable; need to weaken to observable
```

Comment:
I should add the weakening; and, ideally, also allow the user to decorate this as unobservable